src/components/datePicker/DatePickerReducers.ts

```diff
@@ -103,6 +109,17 @@ const resetDates = (state: IDatePickerState, action: IReduxAction<IReduxActionsP
     });
 };
 
+const clearSelection = (state: IDatePickerState, action: IReduxAction<IReduxActionsPayload>): IDatePickerState => {
+  return state.id.indexOf(action.payload.id) !== 0 || !state.isClearable
```

Comment:
What if the index is 0?

Reply:
If the index is zero then it will fall into the second branch of the ternary which is what we want.

Reply:
I'm not sure I understand. It will never apply if the item is the first one in the array? I think the order of the array might always change. Did you mean to check if `index !== -1`? If not, you might want to check for another property that will make sure you target the right items.

Reply:
`state.id` is not an array, it is a string. The logic here is that if the state id doesn't begin with the payload id, then the state is not modified.

This allows to match partial ids like so:
* `MyDatePickerDropdown`
* `MyDatePickerDropdown-calendar`

Reply:
Sorry about that, I get it now. Thanks.